base/gcutils.jl

```diff
@@ -162,6 +162,10 @@ function disable_finalizers() @inline
     ccall(:jl_gc_disable_finalizers_internal, Cvoid, ())
 end
 
+function in_finalizer(t::Task=current_task()) @inline
+    ccall(:jl_gc_is_in_finalizer, Int8, (Any,), t) > 0
+end
```

Comment:
This signature introduces a probably data race on `ptls` as well as a null-pointer dereference on `in_finalizer `: there is no reliable and valid way to check in_finalizer for any task except for `current_task`.

Reply:
Removed the argument in the latest commit.